stellargraph/core/graph.py

```diff
@@ -895,6 +900,18 @@ def str_node_type(count, nt):
                 edge_types = "none"
             return f"{nt}: [{count}]\n    Features: {feature_text}\n    Edge types: {edge_types}"
 
+        def str_weight_info(metrics):
+            if metrics.min == metrics.max:
+                if metrics.min == 1:
+                    return f"Weights: all 1.0 (default)"
+                else:
+                    return f"Weights: all {metrics.min}"
+            else:
+                return f"Weights: range=[{metrics.min}, {metrics.max}], mean={metrics.mean}, std={metrics.std}"
```

Comment:
The test show that some of these can end up with a lot of sig-figs even for simple data; is it worth truncating them similar to how pandas does in its DataFrame view. We could either use a hard coded value, like:

```suggestion
                return f"Weights: range=[{metrics.min:.6}, {metrics.max:.6}], mean={metrics.mean:.6}, std={metrics.std:.6}"
```

or have the value customisable (either as a parameter or piggy back of pandas):

```python
                prec = pd.get_option("display.precision")
                return f"Weights: range=[{metrics.min:.{prec}}, {metrics.max:.{prec}}], mean={metrics.mean:.{prec}}, std={metrics.std:.{prec}}"
```

I'm not sure if this will work with integer weights, and so we may need to either explicitly cast the values, or use the `g` format specifier (to force it to always use the default floating point formatter), like, for the hard-coded option:

```suggestion
                return f"Weights: range=[{metrics.min:.6g}, {metrics.max:.6g}], mean={metrics.mean:.6g}, std={metrics.std:.6g}"
```